### ISSUE TITLE
Internal functional interface in Kotlin format in an uncompilable way

### ISSUE THREAD

**Issue by yonatankarp** [saveourtool/diktat#1775]
If you're just submitting a feature request or question, no need for the above.

When I'm using an internal functional interface, the formatting puts the `fun` keyword in front of the `internal` keyword causing the code to be un-compileable.

For example, the following class:

```kotlin
internal fun interface DragonSlayingStrategy {
    fun execute()
}
```

Will end up as follows after running spotless:

```kotlin
fun internal interface DragonSlayingStrategy {
    fun execute()
}
```

The above code using Spotless Gradle plugin (version `6.22.0`) with this configuration:

```gradle
spotless {
        kotlin {
            target(
                fileTree("${project.rootDir}/$name") {
                    include("**/*.kt")
                },
            )
            diktat("1.2.3").configFile("${rootProject.rootDir}/config/diktat/diktat-analysis.yml")
        }
    }
```

Here you can see the output log of the issue:

```shell
 @@ -3,6 +3,6 @@
/**
·*·Strategy·interface.
·*/
-internal·fun·interface·DragonSlayingStrategy·{
+fun·internal·interface·DragonSlayingStrategy·{
····fun·execute()
}
```

The issue was reported to Spotless and pin-point to `dektat` as can be seen in this Issue: https://github.com/diffplug/spotless/issues/1865

A code example that reproduces the issue can be found here: https://github.com/yonatankarp/spotless-bug-example

Please note that the issue still happens even when all costume configs are disabled.

**Comment by nulls** [comment]
@yonatankarp thanks for reporting the issue. We will look on it

**Comment by diphtongue** [comment]
This problem was solved in diktat-1.2.5. Please, move to diktat-2.0.0. However, if you need spotless, then move to diktat-1.2.5.

**Comment by yonatankarp** [comment]
Thanks @diphtongue ! Seems to solve the problem 🙏 

I'm closing the issue
